src/rust/inetstack/protocols/tcp/tests/simulator.rs

```diff
@@ -349,6 +342,7 @@ impl Simulation {
             },
         }
 
+        self.engine.poll();
```

Comment:
Should we add this here?

Reply:
Because our usual scheduling policy in the libOS level is to call poll immediately after each sys call. Since this PR moves the send to a coroutine, rather than doing it in the synchronous path, to better align with the other libOSes, we need to poll immediately after issuing the push.

Reply:
Ok, I have added a single poll just to the push system call. It also lets us preserve the push pop trace as before.